livecheck/heuristic.rb

```diff
@@ -108,16 +104,19 @@ def version_heuristic(livecheckable, urls, regex = nil)
       rescue TypeError
         nil
       end
-    elsif url =~ %r{(sourceforge\.net|sf\.net)/} && sourceforge_special_cases.none? { |sc| url.include? sc }
-      project_name = url.match(%r{/projects?/(.*?)/})[1]
+    elsif /(sourceforge|sf)\.net/.match?(url) && sourceforge_special_cases.none? { |sc| url.include? sc }
+      project_name = if url.include?("/project")
+        url.match(%r{/projects?/(.*?)/})[1]
+      elsif url.include?(".net/p/")
+        url.match(%r{\.net/p/(.*?)/})[1]
+      else
+        url.match(%r{\.net(?::/cvsroot)?/([^/]+)})[1]
+      end
       page_url = "https://sourceforge.net/projects/#{project_name}/rss"
 
-      if Homebrew.args.debug?
-        puts "Possible SourceForge project [#{project_name}] detected" \
-             "at #{url}"
-      end
+      puts "Possible SourceForge project [#{project_name}] detected at #{url}" if Homebrew.args.debug?
 
-      regex ||= %r{/#{project_name}/([a-zA-Z0-9.]+(?:\.[a-zA-Z0-9.]+)*)}i
+      regex ||= %r{/#{project_name}/files/.*?[-_/](\d+(?:[-.]\d+)+)[-_/%.]}i
```

Comment:
```suggestion
      regex ||= %r{url=.+/#{project_name}/files/.*?[-_/](\d+(?:[-\.]\d+)+)[-_/%\.]}i
```

We're matching URLs in the RSS text, so we can reduce the number of matches that livecheck has to work through by restricting matching to `url` attribute values. I tested this modification against all the formulae that the SourceForge strategy could apply to (~500 formulae) and the output was the same except for one formula that needs to be fixed with a livecheckable anyway.

There are a couple dots in the regex as well, when it seems like the intention was to match a period (rather than anything), so we should escape those, if appropriate.

Reply:
Within character classes, [periods don't need to be escaped](https://www.regular-expressions.info/charclass.html).

Reply:
Definitely makes sense, considering the context. The wonders never cease with regexes, ahah.

I updated the suggestion, so it only includes the leading `url=.+` modification now.